### PR TITLE
add test for new union syntax on generic bounds

### LINKED ISSUE #10949: new union syntax (`types.Union`) doesn't work on generic bounds
```py
from typing import TypeVar


T = TypeVar("T", bound=int | str)

def foo(value: T):
    ...
```
https://mypy-play.net/?mypy=latest&python=3.10&flags=show-column-numbers%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=e32e412ddf26892faaadcf3bda061ad6

### PR DESCRIPTION
**Pull request by DetachHead**
closes #10949

### CODE REVIEW

**Review by hauntsaninja — APPROVED** [review]
Great, thank you very much! :-)

**Comment by DetachHead** [comment]
@hauntsaninja i'm not sure why the 3.10 tests are failing because it works when i run the same code locally. i assume 3.9 and below are failing because this new union syntax is only enabled in 3.10, so i moved the test to the newly created `check-python310.test`

**Comment by JelleZijlstra** [comment]
This looks like a fixture issue to me. You may need a different one. See https://github.com/python/mypy/blob/master/test-data/unit/README.md#fixtures for an intro to fixtures.

**Comment by DetachHead** [comment]
i can't seem to find a fixture that works, i thought maybe `[builtins fixtures/type.pyi]` since other tests that use the new union syntax use it but it doesn't seem to make a difference

File: mypy/test/testcheck.py [hunk 1]
```diff
@@ -104,6 +104,8 @@
     typecheck_files.append('check-python38.test')
 if sys.version_info >= (3, 9):
     typecheck_files.append('check-python39.test')
+if sys.version_info >= (3, 10):
```
Comment (sobolevn) [inline mypy/test/testcheck.py:107]: You don't have to create a separate file for just a single test.
You can use `# flags: --python-version 3.10`

File: test-data/unit/check-python310.test [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    c: int
+
+T = TypeVar("T", bound=U | V | W)
+
+def f(x: T) -> None:
+    x.a  # E
+    x.b = 1  # E
+    del x.c  # E
+
+[builtins fixtures/type.pyi]
```
Comment (sobolevn) [inline test-data/unit/check-python310.test:18]: Try `[builtins fixtures/tuple.pyi]`

File: mypy/test/testcheck.py [hunk 3]
```diff
@@ -104,6 +104,8 @@
...
+    typecheck_files.append('check-python310.test')
```
Comment (sobolevn) [inline mypy/test/testcheck.py:108]: Moreover, there's already `test-data/unit/check-union-or-syntax.test` file

**Review by sobolevn — APPROVED** [review]
LGTM, thank you! 👍

**Pull request merged**